packages/php-wasm/logger/src/lib/logger.ts

```diff
@@ -108,6 +115,34 @@ export class Logger extends EventTarget {
 		this.windowConnected = true;
 	}
 
+	/**
+	 * Register a listener for service worker messages and log the data.
+	 * The service worker will send the number of open Playground tabs.
+	 */
+	public addServiceWorkerMessageListener() {
+		const getServiceWorkerMetrics = () => {
+			if (!navigator.serviceWorker.controller) {
+				return;
+			}
+			navigator.serviceWorker.controller.postMessage(
+				'getServiceWorkerMetrics'
+			);
+		};
+		getServiceWorkerMetrics();
+		navigator.serviceWorker.addEventListener(
+			'controllerchange',
+			getServiceWorkerMetrics
+		);
+		navigator.serviceWorker.addEventListener('message', (event) => {
+			if (event.data.numberOfOpenPlaygroundTabs) {
```

Comment:
Just in case this is 0 for whatever reason or `data` isn't an object

```suggestion
			if (event.data?.numberOfOpenPlaygroundTabs !== undefined) {
```